examples/deepcopy-gen/generators/deepcopy.go

```diff
@@ -578,93 +618,96 @@ func (g *genDeepCopy) doBuiltin(t *types.Type, sw *generator.SnippetWriter) {
 		default:
 			sw.Do("for key, val := range *in {\n", nil)
-			if g.copyableAndInBounds(t.Elem) {
-				sw.Do("newVal := new($.|raw$)\n", t.Elem)
+			if g.copyableAndInBounds(uet) {
+				sw.Do("newVal := new($.|raw$)\n", ut.Elem)
 				sw.Do("val.DeepCopyInto(newVal)\n", nil)
 				sw.Do("(*out)[key] = *newVal\n", nil)
-			} else if t.Elem.Kind == types.Slice && t.Elem.Elem.Kind == types.Builtin {
+			} else if uet.Kind == types.Slice || uet.Kind == types.Builtin {
 				sw.Do("if val==nil { (*out)[key]=nil } else {\n", nil)
-				sw.Do("(*out)[key] = make($.|raw$, len(val))\n", t.Elem)
+				sw.Do("(*out)[key] = make($.|raw$, len(val))\n", uet)
 				sw.Do("copy((*out)[key], val)\n", nil)
 				sw.Do("}\n", nil)
-			} else if t.Elem.Kind == types.Alias && t.Elem.Underlying.Kind == types.Slice && t.Elem.Underlying.Elem.Kind == types.Builtin {
-				sw.Do("(*out)[key] = make($.|raw$, len(val))\n", t.Elem)
-				sw.Do("copy((*out)[key], val)\n", nil)
-			} else if t.Elem.Kind == types.Pointer {
+			} else if uet.Kind == types.Pointer {
 				sw.Do("if val==nil { (*out)[key]=nil } else {\n", nil)
-				sw.Do("(*out)[key] = new($.Elem|raw$)\n", t.Elem)
+				sw.Do("(*out)[key] = new($.Elem|raw$)\n", uet)
 				sw.Do("val.DeepCopyInto((*out)[key])\n", nil)
 				sw.Do("}\n", nil)
 			} else {
-				sw.Do("(*out)[key] = *val.DeepCopy()\n", t.Elem)
+				sw.Do("(*out)[key] = *val.DeepCopy()\n", uet)
 			}
 			sw.Do("}\n", nil)
 		}
 	} else {
 		// TODO: Implement it when necessary.
 		sw.Do("for range *in {\n", nil)
-		sw.Do("// FIXME: Copying unassignable keys unsupported $.|raw$\n", t.Key)
+		sw.Do("// FIXME: Copying unassignable keys unsupported $.|raw$\n", ut.Key)
 		sw.Do("}\n", nil)
 	}
 }
 
 func (g *genDeepCopy) doSlice(t *types.Type, sw *generator.SnippetWriter) {
+	ut := underlyingType(t)
+	uet := underlyingType(ut.Elem)
+
 	if hasDeepCopyMethod(t) {
 		sw.Do("*out = in.DeepCopy()\n", nil)
 		return
 	}
 
 	sw.Do("*out = make($.|raw$, len(*in))\n", t)
-	if hasDeepCopyMethod(t.Elem) {
+	if hasDeepCopyMethod(ut.Elem) {
 		sw.Do("for i := range *in {\n", nil)
 		sw.Do("(*out)[i] = (*in)[i].DeepCopy()\n", nil)
 		sw.Do("}\n", nil)
-	} else if t.Elem.Kind == types.Builtin || t.Elem.IsAssignable() {
+	} else if uet.Kind == types.Builtin || uet.IsAssignable() {
 		sw.Do("copy(*out, *in)\n", nil)
 	} else {
 		sw.Do("for i := range *in {\n", nil)
-		if t.Elem.Kind == types.Slice {
+		if uet.Kind == types.Slice {
 			sw.Do("if (*in)[i] != nil {\n", nil)
 			sw.Do("in, out := &(*in)[i], &(*out)[i]\n", nil)
-			g.generateFor(t.Elem, sw)
+			g.generateFor(ut.Elem, sw)
 			sw.Do("}\n", nil)
-		} else if hasDeepCopyMethod(t.Elem) {
+		} else if hasDeepCopyMethod(ut.Elem) {
 			sw.Do("(*out)[i] = (*in)[i].DeepCopy()\n", nil)
 			// REVISIT(sttts): the following is removed in master
 			//} else if t.Elem.IsAssignable() {
 			//	sw.Do("(*out)[i] = (*in)[i]\n", nil)
-		} else if t.Elem.Kind == types.Interface {
+		} else if ut.Elem.Kind == types.Interface {
```

Comment:
Same here, wrt `uet.Kind` instead of `ut.Elem.Kind`?

Reply:
dito